packages/core/src/event_delegation_utils.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  EarlyJsactionDataContainer,
+  EventContract,
+  EventContractContainer,
+  EventDispatcher,
+  JSACTION,
+  registerDispatcher,
+} from '@angular/core/primitives/event-dispatch';
+import {Injectable, Injector, inject} from './di';
+import {APP_ID} from './application/application_tokens';
+
+declare global {
+  var ngContracts: {[key: string]: EarlyJsactionDataContainer};
+  interface Element {
+    __jsaction_fns: Map<string, Function[]> | undefined;
+  }
+}
+
+export const CONTRACT_PROPERTY = 'ngContracts';
+
+export function handleEvent(event: Event) {
```

Comment:
nit: we can give this function a bit more descriptive name, for example:
```suggestion
export function invokeRegisteredListeners(event: Event) {
```

Reply:
done